api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/data/implementation/ResourceGenerator.java

```diff
@@ -0,0 +1,134 @@
+	 * part name is {@value #JAVA_UUID}, then new generated UUID will be used as a
+	 * part If part name is {@value #UUID_NUMBER}, then new generated UUID
+	 * represented as a string of 40 numbers will be used as a part If part name is
+	 * {@value #OLD_NUMBER} and it is the last one, then old individual uri should
+	 * be created. All previously defined parts will be used as a new namespace.
+	 * @throws InitializationException 
+	 */
+
+	public Resource getUriFromFormat(String input) throws InitializationException {
+		String uri = getUri(input);
+		Resource res = new ResourceImpl(uri);
+		return res;
+	}
+
+	private String getUri(String input) throws InitializationException {
+		String[] parts = input.split("[\\s,]+");
+		StringBuilder uriBuilder = new StringBuilder();
+		for (int i = 0; i < parts.length; i++) {
+			String partName = parts[i].trim();
+			if (isOldFormatRequested(parts, i, partName) && wadf != null) {
+				return getOldURI(uriBuilder.toString());
+			}
+			uriBuilder.append(getUriPart(partName));
+		}
+		if (uriBuilder.length() == 0) {
+			uriBuilder.append(getJavaUUID());
+		}
+		String uri = uriBuilder.toString();
+		if (wadf != null && wadf.checkURI(uri) != null) {
+			throw new InitializationException("Uri check failed");
+		}
+		return uri;
+	}
+
+	private boolean isOldFormatRequested(String[] parts, int i, String part) {
+		return parts.length == i + 1 && OLD_NUMBER.equals(part);
+	}
+
+	private static String getUriPart(String keyword) {
+		if (JAVA_UUID.equals(keyword)) {
+			return getJavaUUID();
+		}
+		if (UUID_NUMBER.equals(keyword)) {
+			return getJavaUUIDNumber();
+		}
+		//If wadf is not available fallback to use of UUID_NUMBER
+		if (OLD_NUMBER.equals(keyword)) {
+			return getJavaUUIDNumber();
+		}
+		return keyword;
+	}
+
+	private static String getJavaUUIDNumber() {
+		final UUID randomUUID = UUID.randomUUID();
+		final String base16String = randomUUID.toString().replace("-", "");
+		return String.format("%040d", new BigInteger(base16String, 16));
+	}
+
+	private static String getJavaUUID() {
+		return UUID.randomUUID().toString();
+	}
+
+	public void init(WebappDaoFactory webappDaoFactory) {
+		wadf = webappDaoFactory;
+	}
+
+	private String getOldURI(String newNamespace) throws InitializationException {
+		String uri = null;
+		String errMsg = null;
+		Random random = new Random();
+		boolean uriIsGood = false;
+		int attempts = 0;
+
+		while (!uriIsGood && attempts < 30) {
+			uri = newNamespace + random.nextInt(Math.min(Integer.MAX_VALUE, (int) Math.pow(2, attempts + 13)));
+			errMsg = wadf.checkURI(uri);
+			if (errMsg != null) {
+				uri = null;
+			} else {
+				uriIsGood = true;
+			}
+			attempts++;
+		}
+		if (uri == null) {
+			log.error("Generated uri is null, fallback to generate uri in new format");
+			uri = newNamespace + getJavaUUIDNumber();
+		}
+		if (wadf.checkURI(uri) != null) {
+			throw new InitializationException("Uri check failed");
+		}
+
+		return uri;
+	}
```

Comment:
```suggestion
	private String getUri(String input) throws InitializationException {
		String[] parts = input.split("[\\s,]+");
		StringBuilder uriBuilder = new StringBuilder();
		for (int i = 0; i < parts.length; i++) {
			String partName = parts[i].trim();
			uriBuilder.append(getUriPart(uriBuilder.toString(), partName));
		}
		if (uriBuilder.length() == 0) {
			uriBuilder.append(getJavaUUID());
		}
		String uri = uriBuilder.toString();
		if (wadf != null && wadf.checkURI(uri) != null) {
			throw new InitializationException("Uri check failed");
		}
		return uri;
	}

	private static String getUriPart(String namespace, String keyword) {
		if (JAVA_UUID.equals(keyword)) {
			return getJavaUUID();
		}
		if (UUID_NUMBER.equals(keyword)) {
			return getJavaUUIDNumber();
		}
		if (OLD_NUMBER.equals(keyword)) {
			return getOldURI(namespace);
		}
		return keyword;
	}

	private static String getJavaUUIDNumber() {
		final UUID randomUUID = UUID.randomUUID();
		final String base16String = randomUUID.toString().replace("-", "");
		return String.format("%040d", new BigInteger(base16String, 16));
	}

	private static String getJavaUUID() {
		return UUID.randomUUID().toString();
	}

	public void init(WebappDaoFactory webappDaoFactory) {
		wadf = webappDaoFactory;
	}

	private String getOldURI(String newNamespace) throws InitializationException {
		if (wadf == null)
		    return getJavaUUIDNumber();
		String uri = null;
		String uriNumber = null;
		String errMsg = null;
		Random random = new Random();
		boolean uriIsGood = false;
		int attempts = 0;

		while (!uriIsGood && attempts < 30) {
			uriNumber = random.nextInt(Math.min(Integer.MAX_VALUE, (int) Math.pow(2, attempts + 13)));
			uri = newNamespace + uriNumber;
			errMsg = wadf.checkURI(uri);
			if (errMsg != null) {
				uriNumber = null;
				uri = null;
			} else {
				uriIsGood = true;
			}
			attempts++;
		}
		if (uriNumber == null) {
			log.error("Generated uri is null, fallback to generate uri in new format");
			uriNumber = getJavaUUIDNumber();
		}
		/*
		I think this is not needed, because it is checked at the end of the getUri method
		if (wadf.checkURI(uri) != null) {
			throw new InitializationException("Uri check failed");
		}*/

		return uriNumber;
	}
```

Reply:
This solution still force us to mix new approach of constructing URIs from user defined schemes and old approach.
I think I made a mistake trying to mix them. 
So I suggest to drop support of old approach, as it is not only rigid, but also more expensive and produces less unique uri's.
We should decide what should be our standard, but I think it makes sense to leave this easy configurable.